File: src/get_token.cpp
#include <iostream>
#include <memory>

#include "gtoken.hpp"

int main() {
    const std::string text = "Hallo Welt";

    auto acquirer =
        std::make_unique<TokenAcquirer>("0", "https://translate.google.com");
    const std::string& token = acquirer->do_token(text);
    std::cout << "Token:" << token << std::endl;
    return 0;
}

File: get_token.py
import asyncio
import httpx
import sys
import os

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "py-googletrans"))

from googletrans.gtoken import TokenAcquirer
from googletrans import Translator

text = "Hallo Welt"

async def main():
    async with httpx.AsyncClient(http2=True) as client:
        acquirer = TokenAcquirer(client=client)
        token = await acquirer.do(text)
        print(f"Acquired token: {token}")

    async with Translator() as translator:
        result = await translator.translate(text, dest="fr")
        print(f"Translated: \"{text}\" --> \"{result.text}\"")

if __name__ == "__main__":
    asyncio.run(main())
